users/engler/fairseq/training.py

```diff
@@ -232,6 +235,69 @@ def run(self):
             my_env["PYTHONPATH"] = self.fairseq_root
         sp.check_call(self._get_run_cmd(), env=my_env)
 
+    def plot(self):
+        directory = "./outputs"
+        train_score, train_error = {}, {}
+        valid_score, valid_error = {}, {}
+
+        for cur in os.walk(directory):
+            dir_path = cur[0]
+            files = cur[2]
+            if 'hydra_train.log' in files:
+                with open(f'{dir_path}/hydra_train.log', 'r') as f:
+                    while True:
+                        line = f.readline()
+                        if 'begin validation on "valid" subset' in line:
+                            split = re.sub('[{,":]', '', f.readline()).split(' ')
+                            try:
+                                epoch = int(split[split.index('epoch') + 1])
+                                score = float(split[split.index('valid_loss') + 1])
+                                error = 1 - float(split[split.index('valid_accuracy') + 1])
+                            except ValueError:
+                                continue
```

Comment:
Yes, it can happen that the training crashes directly after logging the line `begin validation on "valid" subset`. Then the next line is empty and we get a `ValueError`. In this case we can just continue because the required values will be logged in the log file for the restarted training again. I think I should make this check more explicit by checking `f.readline()` being an empty string instead of catching the ValueError.

Reply:
Yes, let's keep it but please add a brief comment on that.